### PR TITLE
fix:when SetRetryCount -1, both resp and error are nil.

### PR DESCRIPTION
**Pull request by yjhmelody**
I add a test for this behavior, I think it is a bug.

### CODE REVIEW

**Review by jeevatkm — COMMENTED** [review]
@yjhmelody Thanks for creating a PR. I'm sorry I'm not able to grasp the outcome.

Can you explain and help me to understand?

**Comment by yjhmelody** [comment]
@jeevatkm 

https://github.com/go-resty/resty/blob/master/retry.go#L87

```go
for attempt := 0; attempt <= opts.maxRetries; attempt++ {
...
```
opts.MaxRetries is `-1` so we cannot go into the loop,
The `Backoff`'s all options have not done anything exactly.

File: client_test.go [hunk 1]
```diff
@@ -612,118 +612,3 @@ func TestNewWithLocalAddr(t *testing.T) {
 	assertNil(t, err)
 	assertEqual(t, resp.String(), "TestGet: text response")
 }
-
-func TestClientOnResponseError(t *testing.T) {
```
Comment (moorereason) [inline client_test.go:616]: Why are these two test functions removed?

Comment (yjhmelody) [reply client_test.go:616]: I'm sorry that I do a wrong rebased operation. I will fix it.

**Comment by jeevatkm** [comment]
@yjhmelody This PR contains new test cases added. But you mentioned it as "fix". What is the bug and fix you're submitting?

**Comment by moorereason** [comment]
I created issue #404 to track the actual issue reported in the PR.

**Comment by yjhmelody** [comment]
@moorereason Oh, let we choose the second solution ?

**Review by jeevatkm — APPROVED** [review]
@yjhmelody Thanks for your contributions. I'm really sorry for the delay, due to personal reasons.

**Pull request merged**